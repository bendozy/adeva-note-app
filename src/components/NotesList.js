import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import Button from '@material-ui/core/Button';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

const useStyles = makeStyles({
  search: {
    marginTop: 5,
    display: 'flex',
    alignItems: 'center',
    width: '100%',
  },
  list: {
    // height: '93vh',
  },
  input: {
    marginLeft: 8,
    flex: 1,
  },
  iconButton: {
    padding: 10,
  },
});

const NotesList = ({ notes }) => {
  const classes = useStyles();
  const [searchText, setSearchText] = useState('');

  return (
    <Grid>
      <Grid>
        <Paper className={classes.search}>
          <IconButton className={classes.iconButton} aria-label="Menu">
            <MenuIcon />
          </IconButton>
          <InputBase
            className={classes.input}
            placeholder="Search Notes"
            inputProps={{ 'aria-label': 'Search Notes' }}
            value={searchText}
            onChange={event => setSearchText(event.target.value)}
          />
          <IconButton className={classes.iconButton} aria-label="Search">
            <SearchIcon />
          </IconButton>
        </Paper>
      </Grid>
      <Grid>
        <List component="nav" aria-label="Secondary mailbox folders">
          {notes
            .filter(note =>
              note.title.toLowerCase().includes(searchText.toLowerCase())
                || note.details.toLowerCase().includes(searchText.toLowerCase()))
            .map((note, index) => (
              <ListItem
                key={note.id}
                button
                divider={notes.length !== index + 1}
                component={Link}
                to={`/notes/${note.id}`}
              >
                <ListItemText primary={note.title} />
              </ListItem>
            ))
          }
        </List>
      </Grid>
      <Grid>
        <Button
          fullWidth
          variant="contained"
          color="primary"
          component={Link}
          to="/notes/add"
        >
          Add New Note
        </Button>
      </Grid>
    </Grid>
  );
};

NotesList.defaultProps = {
  notes: [],
};

NotesList.propTypes = {
  notes: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    details: PropTypes.string.isRequired,
  })),
};

const mapStateToProps = ({ notes }) => ({ notes });

export default connect(mapStateToProps)(NotesList);
