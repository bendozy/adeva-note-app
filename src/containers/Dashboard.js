import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
} from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import HomeIcon from '@material-ui/icons/Home';
import Paper from '@material-ui/core/Paper';
import CssBaseline from '@material-ui/core/CssBaseline';
import Grid from '@material-ui/core/Grid';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import { makeStyles } from '@material-ui/core/styles';
import Home from '../components/Home';
import Note from '../components/Note';
import NoteForm from '../components/NoteForm';
import NotesList from '../components/NotesList';
import EmptyPage from '../components/EmptyPage';

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    backgroundColor: 'white',
  },
  homeIconButton: {
    color: 'white',
  },
  paper: {
    padding: theme.spacing(2),
    height: '100vh',
  },
}));

const Dashboard = () => {
  const classes = useStyles();

  return (
    <Router>
      <React.Fragment>
        <CssBaseline />
        <AppBar position="relative">
          <Toolbar>
            <IconButton
              className={classes.homeIconButton}
              aria-label="Home"
              component={Link}
              to="/"
            >
              <HomeIcon />
            </IconButton>
            <Typography variant="h6" color="inherit" noWrap>
              Adeva Note App
            </Typography>
          </Toolbar>
        </AppBar>
        <main>
          <Grid container spacing={2} className={classes.root}>
            <Grid item xs={12} sm={3}>
              <NotesList />
            </Grid>
            <Grid item xs={12} sm={9}>
              <Paper className={classes.paper}>
                <Switch>
                  <Route exact path="/" component={Home} />
                  <Route exact path="/notes/add" component={NoteForm} />
                  <Route exact path="/notes/:noteId" component={Note} />
                  <Route exact path="/notes/:noteId/edit" component={NoteForm} />
                  <Route component={EmptyPage} />
                </Switch>
              </Paper>
            </Grid>
          </Grid>
        </main>
      </React.Fragment>
    </Router>
  );
};

export default Dashboard;
