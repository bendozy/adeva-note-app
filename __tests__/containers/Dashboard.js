import React from 'react';
import toJson from 'enzyme-to-json';
import AppBar from '@material-ui/core/AppBar';
import Dashboard from '../../src/containers//Dashboard';
import NotesList from '../../src/components/NotesList';

describe('<Dashboard />', () => {
  it('should render a <Dashboard /> component', () => {
    const wrapper = shallow(<Dashboard />);

    expect(wrapper.find(AppBar)).toHaveLength(1);
    expect(wrapper.find(NotesList)).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
