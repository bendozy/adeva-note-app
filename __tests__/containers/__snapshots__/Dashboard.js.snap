// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render a <Dashboard /> component 1`] = `
<BrowserRouter>
  <CssBaseline />
  <WithStyles(ForwardRef(AppBar))
    position="relative"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Home"
        className="makeStyles-homeIconButton-2"
        component={[Function]}
        to="/"
      >
        <HomeIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        noWrap={true}
        variant="h6"
      >
        Adeva Note App
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <main>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-root-1"
      container={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={3}
        xs={12}
      >
        <Connect(NotesList) />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={9}
        xs={12}
      >
        <WithStyles(ForwardRef(Paper))
          className="makeStyles-paper-3"
        >
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(NoteForm)",
                  "type": [Function],
                }
              }
              exact={true}
              path="/notes/add"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Note)",
                  "type": [Function],
                }
              }
              exact={true}
              path="/notes/:noteId"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(NoteForm)",
                  "type": [Function],
                }
              }
              exact={true}
              path="/notes/:noteId/edit"
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </WithStyles(ForwardRef(Paper))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </main>
</BrowserRouter>
`;
